File: 백준/Silver/10866. 덱/덱.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Deque;
import java.util.LinkedList;

public class Main {
    public static void main(String[] args) throws NumberFormatException, IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        Deque<Integer> dq = new LinkedList<>();
        StringBuffer sb = new StringBuffer();
        for (int i = 0; i < n; i++) {
            String[] op = br.readLine().split(" ");
            switch (op[0]) {
                case "push_front":
                    int x1 = Integer.parseInt(op[1]);
                    dq.addFirst(x1);
                    break;
                case "push_back":
                    int x2 = Integer.parseInt(op[1]);
                    dq.addLast(x2);
                    break;
                case "pop_front":
                    if (dq.isEmpty()) {
                        sb.append(-1).append("\n");
                    } else {
                        sb.append(dq.pollFirst()).append("\n");
                    }
                    break;
                case "pop_back":
                    if (dq.isEmpty()) {
                        sb.append(-1).append("\n");
                    } else {
                        sb.append(dq.pollLast()).append("\n");
                    }
                    break;
                case "size":
                    sb.append(dq.size()).append("\n");
                    break;
                case "empty":
                    if (dq.isEmpty()) {
                        sb.append(1).append("\n");
                    } else {
                        sb.append(0).append("\n");
                    }
                    break;
                case "front":
                    if (dq.isEmpty()) {
                        sb.append(-1).append("\n");
                    } else {
                        sb.append(dq.getFirst()).append("\n");
                    }
                    break;
                case "back":
                    if (dq.isEmpty()) {
                        sb.append(-1).append("\n");
                    } else {
                        sb.append(dq.getLast()).append("\n");
                    }
                    break;
            }
        }
        System.out.println(sb);
    }
}

File: 백준/Silver/10866. 덱/덱.py
from sys import stdin
from collections import deque
input = stdin.readline


def solve():
    N = int(input())

    dq = deque()
    for _ in range(N):
        op = input().split()
        if op[0] == 'push_front':
            x = int(op[1])
            dq.appendleft(x)
        elif op[0] == 'push_back':
            x = int(op[1])
            dq.append(x)
        elif op[0] == 'pop_front':
            if dq:
                print(dq.popleft())
            else:
                print(-1)
        elif op[0] == 'pop_back':
            if dq:
                print(dq.pop())
            else:
                print(-1)
        elif op[0] == 'size':
            print(len(dq))
        elif op[0] == 'empty':
            if dq:
                print(0)
            else:
                print(1)
        elif op[0] == 'front':
            if dq:
                print(dq[0])
            else:
                print(-1)
        else:
            if dq:
                print(dq[-1])
            else:
                print(-1)


if __name__ == "__main__":
    solve()
